apps/remix-ide/src/app/components/plugin-manager-component.js

```diff
@@ -102,6 +102,57 @@ class PluginManagerComponent extends ViewPlugin {
     return this.appManager.actives.includes(name)
   }
 
+  async registerFlattener (event) {
+    if (!this.appManager.isActive('solidity')) { await this.appManager.activatePlugin('solidity') }
+    const compiledSuccessfully = await this.call('solidity', 'compileFile', event)
+    const reportCompileIssue = async () => {
+      this.call('fileManager', 'open', event.path[0])
+      addToolTip('Cannot flatten the file. Please make sure it is compiling successfully.')
+      await this.call('menuicons', 'select', 'solidity')
+    }
+    setTimeout(async () => {
```

Comment:
now that you `await` here `await this.call('solidity', 'compileFile', event)` you shouldn't need the `setTimeout` anymore